modules/Sources/Dependencies/AddressBookClient/AddressBookEncryption.swift

```diff
@@ -0,0 +1,286 @@
+    ///     [Encrypted data]        `timestamp`
+    ///     [Encrypted data]        `contacts`
+    static func contactsFrom(encryptedData: Data) throws -> AddressBookContacts {
+        @Dependency(\.walletStorage) var walletStorage
+        
+        guard let encryptionKeys = try? walletStorage.exportAddressBookEncryptionKeys(), let addressBookKey = encryptionKeys.getCached(account: 0) else {
+            throw AddressBookClient.AddressBookClientError.missingEncryptionKey
+        }
+        
+        var offset = 0
+        
+        // Deserialize `encryption version`
+        let encryptionVersionBytes = encryptedData.subdata(in: offset..<(offset + MemoryLayout<Int>.size))
+        offset += MemoryLayout<Int>.size
+        
+        guard let encryptionVersion = AddressBookClient.bytesToInt(Array(encryptionVersionBytes)) else {
+            return .empty
+        }
+        
+        if encryptionVersion == AddressBookEncryptionKeys.Constants.version {
+            let encryptedSubData = encryptedData.subdata(in: offset..<encryptedData.count)
+            
+            // Derive the sub-key for decrypting the address book.
+            let salt = encryptedSubData.prefix(upTo: 32)
+            let subKey = addressBookKey.deriveEncryptionKey(salt: salt)
+            
+            // Unseal the encrypted address book.
+            let sealed = try ChaChaPoly.SealedBox.init(combined: encryptedSubData.suffix(from: 32))
+            let data = try ChaChaPoly.open(sealed, using: subKey)
+            offset = 0
+            
+            // Deserialize `address book version`
+            let abVersionCountBytes = data.subdata(in: offset..<(offset + MemoryLayout<Int>.size))
+            guard let abVersion =  AddressBookClient.bytesToInt(Array(abVersionCountBytes)) else {
+                return .empty
+            }
+            offset += MemoryLayout<Int>.size
+            
+            // Deserialize `lastUpdated`
+            guard let lastUpdated = AddressBookClient.deserializeDate(from: data, at: &offset) else {
+                return .empty
+            }
+            
+            // Deserialize `contactsCount`
+            let contactsCountBytes = data.subdata(in: offset..<(offset + MemoryLayout<Int>.size))
+            offset += MemoryLayout<Int>.size
+            
+            guard let contactsCount = AddressBookClient.bytesToInt(Array(contactsCountBytes)) else {
+                return .empty
+            }
+            
+            var contacts: [Contact] = []
+            for _ in 0..<contactsCount {
+                if let contact = AddressBookClient.deserializeContact(from: data, at: &offset) {
+                    contacts.append(contact)
+                }
+            }
+            
+            let abContacts = AddressBookContacts(
+                lastUpdated: lastUpdated,
+                version: abVersion,
+                contacts: IdentifiedArrayOf(uniqueElements: contacts)
+            )
```

Comment:
This duplicates `contactsFrom(unencryptedData: Data)`. It should just call it.

Reply:
This is blocking because parsing code especially shouldn't be duplicated.

Reply:
```swift
return try contactsFrom(unencryptedData: data)
```

used inside instead